composeApp/src/commonMain/kotlin/sp/bvantur/tasky/register/presentation/RegisterViewModel.kt

```diff
@@ -0,0 +1,120 @@
+package sp.bvantur.tasky.register.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import sp.bvantur.tasky.core.DispatcherProvider
+import sp.bvantur.tasky.core.domain.ValidateEmailUseCase
+import sp.bvantur.tasky.core.domain.ValidateNameUseCase
+import sp.bvantur.tasky.core.domain.ValidatePasswordUseCase
+import sp.bvantur.tasky.core.presentation.ViewModelUserActionHandler
+import sp.bvantur.tasky.core.presentation.ViewModelViewStateHandler
+import sp.bvantur.tasky.core.presentation.ViewModelViewStateHandlerImpl
+import kotlin.time.Duration.Companion.milliseconds
+
+typealias OnRegisterUserAction = (RegisterUserAction) -> Unit
+
+@OptIn(FlowPreview::class)
+class RegisterViewModel(
+    private val dispatcherProvider: DispatcherProvider,
+    private val validateNameUseCase: ValidateNameUseCase,
+    private val validateEmailUseCase: ValidateEmailUseCase,
+    private val validatePasswordUseCase: ValidatePasswordUseCase
+) : ViewModel(),
+    ViewModelUserActionHandler<RegisterUserAction>,
+    ViewModelViewStateHandler<RegisterViewState> by ViewModelViewStateHandlerImpl(
+        RegisterViewState(),
+        dispatcherProvider
+    ) {
+
+    private val textDebounceFlow = MutableSharedFlow<Pair<String, suspend (String) -> Unit>>()
+
+    init {
+        textDebounceFlow
```

Comment:
I would consider this premature optimization, since you probably won't ever find a more or less modern device where validation an email after a keystroke leads to a noticeable performance issue

Reply:
Ok, I am removing it.